pkg/gitrepo/repo.go

```diff
@@ -303,6 +308,77 @@ func (r *Repo) ResolveVersion(ctx context.Context, ref string) (string, error) {
 	return pseudoVersion, nil
 }
 
+// GetProjectVersion retrieves version stored in project's Go source code.
+// It looks up the value of variable `version` in `pkg/project/project.go` file
+// on master branch.
+func (r *Repo) GetProjectVersion() (string, error) {
+	fileName := "pkg/project/project.go"
+	varName := "version"
+
+	repo, err := git.Open(r.storage, r.worktree)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	worktree, err := repo.Worktree()
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	// When empty CheckoutOptions defaults to master branch.
+	err = worktree.Checkout(&git.CheckoutOptions{})
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	file, err := worktree.Filesystem.Open(fileName)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
```

Comment:
It would be good to handle non-existing pkg/project/project.go file somewhat nicer here.

Reply:
Do you mean having a specific error for this ?

Reply:
Let's discuss when it's split. When `GetFile` has `IsNotFound` matcher then this is fine.